### PR TITLE
Revert "Merge pull request #3336 from rtfd/use-active-for-stable"

### PR DESCRIPTION
**Pull request by ericholscher**
This reverts commit 0d9c3fefb1ea8a64272bfeff42cb37b8122ae335, reversing
changes made to 73b693ce4dd408e6a9f2812a8b7aaa9b9d205c7a.

### CODE REVIEW

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -707,7 +707,7 @@ def update_stable_version(self):
             if current_stable:
                 identifier_updated = (
                     new_stable.identifier != current_stable.identifier)
-                if identifier_updated and current_stable.active:
+                if identifier_updated and current_stable.active and current_stable.machine:
```
Comment (humitos) [inline readthedocs/projects/models.py]: So, comparing this with the original code, we are just adding a another condition (has to be `active`) but we keep the `machine`.

In other words, we are just updating the stable version only if, the identifier changed (a new tag become the stable, for example) and the stable version is active and it was created by RTD.

If what I'm saying is correct, I think we are OK :)

File: readthedocs/rtd_tests/tests/test_sync_versions.py [hunk 2]
```diff
@@ -391,7 +388,7 @@ def test_update_inactive_stable_version(self):
 
         version_stable = Version.objects.get(slug=STABLE)
         self.assertFalse(version_stable.active)
-        self.assertEqual(version_stable.identifier, '0.9')
+        self.assertEqual(version_stable.identifier, '1.0.0')
```
Comment (agjohnson) [inline readthedocs/rtd_tests/tests/test_sync_versions.py]: Shouldn't this test be for `0.9`? Or is this test being repurposed to check the new logic?

File: readthedocs/projects/models.py (hunk 1 above)
Comment (ericholscher) [reply readthedocs/projects/models.py]: Yep, thats the goal :)

File: readthedocs/rtd_tests/tests/test_sync_versions.py (hunk 2 above)
Comment (ericholscher) [reply readthedocs/rtd_tests/tests/test_sync_versions.py]: Yea, that was the original fix.

File: readthedocs/rtd_tests/tests/test_sync_versions.py [hunk 3]
```diff
@@ -468,3 +468,79 @@ def test_unicode(self):
...
+        Version.objects.create(
+            project=self.pip,
+            identifier='0.8.3',
+            verbose_name='0.8.3',
+            active=True,
+        )
+
+        Version.objects.create(
+            project=self.pip,
+            identifier='foo',
```
Comment (humitos) [inline readthedocs/rtd_tests/tests/test_sync_versions.py]: `origin/stable` here I guess.

Comment (ericholscher) [reply readthedocs/rtd_tests/tests/test_sync_versions.py]: I'm tested it gets updated from a prior "set" version.

**Comment by ericholscher** [comment]
Ready for re-review.

**Pull request merged**